### PR TITLE
Change fields in eval_hessian_lagrangian and product to support views

### PR DESCRIPTION
**Pull request by abelsiqueira**
Port of #1548.

### CODE REVIEW

File: src/nlp.jl [hunk 1]
```diff
@@ -659,7 +659,7 @@ end
 function MOI.eval_hessian_lagrangian(
     d::NLPEvaluator,
     H::AbstractVector{Float64},      # Sparse hessian entry vector
-    x::Vector{Float64},              # Current solution
+    x::AbstractVector{Float64},              # Current solution
```
Comment (mlubin) [inline src/nlp.jl]: Nit: alignment of comments

File: test/nlp.jl [hunk 2]
```diff
@@ -168,6 +168,19 @@
         MOI.eval_hessian_lagrangian(d, V, values, 1.0, Float64[])
         @test isapprox(dense_hessian(hessian_sparsity, V, 3), [0.0 1.0 0.0; 1.0 0.0 0.0; 0.0 0.0 2.0])
 
+        # test view
```
Comment (mlubin) [inline test/nlp.jl]: Please create a separate `testset` for testing views.

File: test/nlp.jl [hunk 3]
```diff
@@ -256,6 +269,21 @@
         v = [2.4,3.5,1.2]
         MOI.eval_hessian_lagrangian_product(d, h, values, v, 1.0, [2.0,3.0])
         @test isapprox(h, correct_hessian*v)
+
+        h2 = zeros(10)
```
Comment (mlubin) [inline test/nlp.jl]: Please use a separate testset for this.

**Comment by abelsiqueira** [comment]
Updated with https://github.com/mlubin/ReverseDiffSparse.jl/pull/54 as well.

**Pull request merged**